jdaviz/app.py

```diff
@@ -99,13 +111,69 @@ def to_unit(self, data, cid, values, original_units, target_units):
                 spec = data.get_object(cls=Spectrum1D)
             except RuntimeError:
                 eqv = []
-            else:
-                if len(values) == 2:
-                    # Need this for setting the y-limits
-                    spec_limits = [spec.spectral_axis[0].value, spec.spectral_axis[-1].value]
-                    eqv = u.spectral_density(spec_limits * spec.spectral_axis.unit)
+            if len(values) == 2:
+                # Need this for setting the y-limits
+                spec_limits = [spec.spectral_axis[0].value, spec.spectral_axis[-1].value]
+                eqv = u.spectral_density(spec_limits * spec.spectral_axis.unit)
+
+            # Ensure a spectrum passed through Spectral Extraction plugin
+            elif '_pixel_scale_factor' in spec.meta:
+                # if spectrum data collection item is in Surface Brightness units
+                if u.sr in spec.unit.bases:
+                    # Data item in data collection does not update from conversion/translation.
+                    # App wide orginal data units are used for conversion, orginal_units and
+                    # target_units dicate the conversion to take place.
+                    if (u.sr in u.Unit(original_units).bases) and \
+                       (u.sr not in u.Unit(target_units).bases) and \
+                       (target_units not in ["AB", "bol", "ST"]):  # astropy supported equivalency
+                        # Surface Brightness -> Flux
+                        eqv = [(u.MJy / u.sr,
+                                u.MJy,
+                                lambda x: (x * spec.meta['_pixel_scale_factor']),
+                                lambda x: x)]
+                        
+                        # below is the generalized version
```

Comment:
do we not need the generalized version already?

Reply:
I believe that the equivalency that gets passed to:
https://github.com/spacetelescope/jdaviz/blob/17993097a56eeefe8b50cd2bf979647c1240aff5/jdaviz/app.py#L180
Will be used irregardless of what the actual units for `original_units` and `target_units`. The generalized version  should be used, but it results in more issues. Also, for this commit I was using the specific equivalency for the sake of simplicity in diagnosing bugs with Tom.